datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxDrService.java

```diff
@@ -0,0 +1,273 @@
+                return AttemptResults.breakFor(pollingMessage + " polling cancelled in inmemory store, id: " + sdxCluster.getId());
+            } else {
+                CloudbreakFlowService.FlowState flowState = cloudbreakFlowService.getLastKnownFlowState(sdxCluster);
+                if (RUNNING.equals(flowState)) {
+                    LOGGER.info("{} polling will continue, cluster has an active flow in Cloudbreak, id: {}", pollingMessage, sdxCluster.getId());
+                    return AttemptResults.justContinue();
+                } else {
+                    return getStackResponseAttemptResult(sdxCluster, pollingMessage, flowState);
+                }
+            }
+        } catch (javax.ws.rs.NotFoundException e) {
+            LOGGER.debug("Stack not found on CB side " + sdxCluster.getClusterName(), e);
+            return AttemptResults.breakFor("Stack not found on CB side " + sdxCluster.getClusterName());
+        }
+    }
+
+    private AttemptResult<StackV4Response> getStackResponseAttemptResult(SdxCluster sdxCluster, String pollingMessage, CloudbreakFlowService.FlowState flowState)
+            throws JsonProcessingException {
+        StackV4Response stackV4Response = stackV4Endpoint.get(0L, sdxCluster.getClusterName(), Collections.emptySet());
+        LOGGER.info("Response from cloudbreak: {}", JsonUtil.writeValueAsString(stackV4Response));
+        ClusterV4Response cluster = stackV4Response.getCluster();
+        if (stackAndClusterAvailable(stackV4Response, cluster)) {
+            return sdxDrSucceeded(stackV4Response);
+        } else {
+            if (Status.BACKUP_FAILED.equals(stackV4Response.getStatus()) ||
+                    Status.RESTORE_FAILED.equals(stackV4Response.getStatus())) {
+                LOGGER.info("{} failed for Stack {} with status {}", pollingMessage, stackV4Response.getName(), stackV4Response.getStatus());
+                return sdxDrFailed(sdxCluster, stackV4Response.getStatusReason(), pollingMessage);
+            } else if (Status.BACKUP_FAILED.equals(stackV4Response.getCluster().getStatus()) ||
+                    Status.RESTORE_FAILED.equals(stackV4Response.getCluster().getStatus())) {
+                LOGGER.info("{} failed for Cluster {} status {}", pollingMessage, stackV4Response.getCluster().getName(),
+                        stackV4Response.getCluster().getStatus());
+                return sdxDrFailed(sdxCluster, stackV4Response.getCluster().getStatusReason(), pollingMessage);
+            } else {
+                if (FINISHED.equals(flowState)) {
+                    LOGGER.info("Flow finished, but Backup/Restore is not complete: {}", sdxCluster.getClusterName());
+                    return sdxDrFailed(sdxCluster, "stack is in improper state", pollingMessage);
+                } else {
+                    return AttemptResults.justContinue();
+                }
+            }
+        }
```

Comment:
this if-else thing got out of hand. please refactor parts of into separate method with meaningful names so it would be easier to follow what happens

Reply:
will refactor it.